modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverSetUtils.java

```diff
@@ -188,82 +200,114 @@ private static ValueResolver getParameterValueResolver(String parameterName, Met
-                                                              reflectionCache, muleContext,
-                                                              valueResolverFactory);
-            actualAcceptReferences = false;
-          } else {
-            resolver = new StaticValueResolver(value);
-          }
-          resolverReference
-              .set(getValueResolverFor(parameterName, arrayType, resolveAndinjectIfStatic(resolver, muleContext),
-                                       getDefaultValue(type),
-                                       getExpressionSupport(arrayType), false, modelProperties, actualAcceptReferences));
-        } catch (MuleException e) {
-          throw new MuleRuntimeException(e);
-        }
-      }
+        @Override
+        public void visitArrayType(ArrayType arrayType) {
+          boolean actualAcceptReferences = acceptsReferences;
+          ValueResolver resolver;
+          try {
+            if (value instanceof Collection) {
 
-      @Override
-      public void visitObject(ObjectType objectType) {
-        boolean actualAcceptReferences = acceptsReferences;
-        ValueResolver resolver;
-        try {
-          if (isMap(objectType)) {
-            if (value instanceof Map) {
-              resolver = getParameterValueResolverForMap(parameterName, objectType, (Map) value, reflectionCache,
-                                                         muleContext, valueResolverFactory);
+              resolver = getParameterValueResolverForCollection(parameterName, arrayType, (Collection) value,
+                                                                reflectionCache, muleContext,
+                                                                valueResolverFactory);
               actualAcceptReferences = false;
             } else {
               resolver = new StaticValueResolver(value);
             }
-          } else {
-            Optional<ValueResolver> pojoResolver =
-                getPojoParameterValueResolver(parameterName, objectType, value, reflectionCache,
-                                              muleContext, valueResolverFactory);
-            if (pojoResolver.isPresent()) {
-              resolver = pojoResolver.get();
-              actualAcceptReferences = false;
+            resolverReference
+                .set(getValueResolverFor(parameterName, arrayType, resolveAndinjectIfStatic(resolver, muleContext),
+                                         getDefaultValue(type),
+                                         getExpressionSupport(arrayType), false, modelProperties, actualAcceptReferences,
+                                         valueResolverFactory));
+          } catch (MuleException e) {
+            throw new MuleRuntimeException(e);
+          }
+        }
+
+        @Override
+        public void visitObject(ObjectType objectType) {
+          boolean actualAcceptReferences = acceptsReferences;
+          ValueResolver resolver;
```

Comment:
are you piggy backing a bug fix in here?

Reply:
I think github is showing this lines as changes because there are now indented one tab further due to them being inside an "if". I am not sure is this is what you meant.